### ISSUE TITLE
Table: Frozen Columns not working in v19

### ISSUE THREAD

**Issue by monsieur-ricky** [primefaces/primeng#17229]
### Describe the bug

The Frozen Columns feature (`pFrozenColumn`) isn't working in version 19. This issue can be tested in the official documents:
- Version 17 demo (working): https://v17.primeng.org/table#frozen-columns
- Version 19 demo (not working): https://primeng.org/table#frozen-columns

### Reason for not contributing a PR

- [ ] Lack of time
- [X] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/zrndjve6

### Environment

Angular CLI: 19.0.6
Node: 20.16.0
Package Manager: npm 10.9.2
OS: darwin x64

### Angular version

19.0.6

### PrimeNG version

v19

### Node version

20.16.0

### Browser(s)

Chrome 131.0.6778.205

### Steps to reproduce the behavior

1. Access the official Table component documentation for version 19
2. Go to the Frozen Column demo: https://primeng.org/table#frozen-columns
3. Lock the balance column and horizontally scroll the table to check that the column **isn't fixed**.

### Expected behavior

1. Access the official Table component documentation for version 19
2. Go to the Frozen Column demo: https://v17.primeng.org/table#frozen-columns
3. Lock the balance column and horizontally scroll the table to verify that the column **is fixed**.

**Comment by LubosSremanak** [comment]
Bug #16939 "fixed the issue" by removing the `sticky` property from frozen columns. I think the intention was to only remove the background on frozen columns to make the row's background visible. However, this is not a solution, as sticky columns without a background will overlap text in other columns (see image below).

<img width="797" alt="image" src="https://github.com/user-attachments/assets/4245bb0e-f908-45c1-8c16-808bafaf75dc" />

I propose reverting commit 516e7e9, as it is not a fix but another bug, reopening issue #16939 and close this issue.

---

### Quick Fix

If anyone needs a quick fix, add the following to `styles.scss`:

```css
.p-datatable-scrollable .p-datatable-frozen-column {
  position: sticky;
  background: var(--p-datatable-header-cell-background);
} 

```

**Comment by gauravp-dev** [comment]
@LubosSremanak If I switch to RTL mode it breaks the alignment of table

**Comment by monsieur-ricky** [comment]
> Bug #16939 "fixed the issue" by removing the `sticky` property from frozen columns. I think the intention was to only remove the background on frozen columns to make the row's background visible. However, this is not a solution, as sticky columns without a background will overlap text in other columns (see image below).
> 
> <img alt="image" width="797" src="https://private-user-images.githubusercontent.com/69248396/401334562-4245bb0e-f908-45c1-8c16-808bafaf75dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY0MTEyNDMsIm5iZiI6MTczNjQxMDk0MywicGF0aCI6Ii82OTI0ODM5Ni80MDEzMzQ1NjItNDI0NWJiMGUtZjkwOC00NWMxLThjMTYtODA4YmFmYWY3NWRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA5VDA4MjIyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhOGVmMzZjZTkzMjU0NGVlNjhlNmRiZTAwZmEzZDg0ZjlkODBkYzFkY2E5MTY0MWU1Y2ExMGQyMmFiM2VlMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.P12EPJ5STiJhsBGVswFFKRQoRPjT3sYz60JttxsJz7Q">
> I propose reverting commit [516e7e9](https://github.com/primefaces/primeng/commit/516e7e9c5e8133cb1bd9dd3e0f76b097999a2a5b), as it is not a fix but another bug, reopening issue #16939 and close this issue.
> 
> ### Quick Fix
> If anyone needs a quick fix, add the following to `styles.scss`:
> 
> ```css
> .p-datatable-scrollable .p-datatable-frozen-column {
>   position: sticky;
>   background: var(--p-datatable-header-cell-background);
> } 
> ```

@LubosSremanak thanks for the feedback.
Does reverting the commit also solve this issue? https://github.com/primefaces/primeng/issues/17229#issuecomment-2578946958

**Comment by LubosSremanak** [comment]
> @LubosSremanak If I switch to RTL mode it breaks the alignment of table

Did it work before commit: 516e7e9? Try https://v17.primeng.org/table#frozen-columns.

If it worked before that commit, then my quick fix is not sufficient for RTL. However, if it didn’t work even before, then this RTL issue is unrelated to this fix.

**Comment by LubosSremanak** [comment]
@monsieur-ricky Yes, as I mentioned in the previous comment, this will resolve it.

**Comment by LubosSremanak** [comment]
> @LubosSremanak sorry for the confusion, I pasted the wrong link. I was referring to this issue: [#17224 (comment)](https://github.com/primefaces/primeng/issues/17224#issue-2759720305)

No, this change affects only frozen columns.

**Comment by monsieur-ricky** [comment]
> > @LubosSremanak sorry for the confusion, I pasted the wrong link. I was referring to this issue: [#17224 (comment)](https://github.com/primefaces/primeng/issues/17224#issue-2759720305)
> 
> No, this change affects only frozen columns.

Ok, thanks for the feedback.

**Comment by cetincakiroglu** [comment]
Related issue: https://github.com/primefaces/primeng/issues/17490

This issue has been fixed in v19.0.6
